### PR TITLE
add default monitoringExporter to wls domain spec.

### PR DESCRIPTION
**Pull request by guoyongzhang**
# Description

- Add the default monitoringExporter to the weblogic domain spec if it is not specified.
- Add sample monitoringExporter to the domain specs in bobs-books example.

Fixes VZ-2706

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go [hunk 1]
```diff
@@ -41,13 +42,127 @@ const (
 	destinationRuleKind       = "DestinationRule"
 )
 
+const monitoringExporterData = `
+  {
+    "configuration": {
+      "domainQualifier": true,
+      "metricsNameSnakeCase": true,
+      "queries": [
```
Comment (markxnelson) [inline application-operator/controllers/wlsworkload/weblogicworkload_controller.go:50]: you can't hard code this - you need to let the user tell you what they want.  its ok to have a default, but its not ok to provide no option to specify their own value

**Review by markxnelson — CHANGES_REQUESTED** [review]

File: application-operator/controllers/wlsworkload/weblogicworkload_controller.go (hunk 1 above)
Comment (guoyongzhang) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller.go:50]: This monitoringExporter config is the default.  Users can specify their own config as shown in examples/bobs-books/bobs-books-comp.yaml.

Comment (markxnelson) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller.go:50]: ok is that what line 601 is doing? i missed that, it was not super clear.
i think we should have tests for both cases (present and defaulted)

Comment (guoyongzhang) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller.go:50]: Good point.  I will add the "present" case.

File: application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go [hunk 2]
```diff
@@ -146,7 +180,97 @@ func TestReconcileCreateWebLogicDomain(t *testing.T) {
...
+
+	var mocker = gomock.NewController(t)
+	var cli = mocks.NewMockClient(mocker)
+
+	appConfigName := "unit-test-app-config"
+	componentName := "unit-test-component"
+	labels := map[string]string{oam.LabelAppComponent: componentName, oam.LabelAppName: appConfigName,
+		constants.LabelWorkloadType: constants.WorkloadTypeWeblogic}
+
+	// expect call to fetch existing WebLogic Domain
```
Comment (markxnelson) [inline application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go:214]: nit pick: domain should not have a capital

File: application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go [hunk 3]
```diff
@@ -146,7 +180,97 @@ func TestReconcileCreateWebLogicDomain(t *testing.T) {
...
+		Get(gomock.Any(), types.NamespacedName{Namespace: namespace, Name: "unit-test-verrazzano-weblogic-workload"}, gomock.Not(gomock.Nil())).
+		DoAndReturn(func(ctx context.Context, name types.NamespacedName, workload *vzapi.VerrazzanoWebLogicWorkload) error {
+			workload.Spec.Template = runtime.RawExtension{Raw: []byte(strings.ReplaceAll(strings.ReplaceAll(weblogicDomainWithMonitoringExporter, " ", ""), "\n", ""))}
+			workload.ObjectMeta.Labels = labels
+			workload.APIVersion = vzapi.SchemeGroupVersion.String()
+			workload.Kind = "VerrazzanoWebLogicWorkload"
+			workload.Namespace = namespace
+			return nil
+		})
+	// expect a call to list the FLUENTD config maps
```
Comment (markxnelson) [inline application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go:231]: is fluentd a copy/paste error here?

**Review by markxnelson — APPROVED** [review]

File: application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go (hunk 3 above)
Comment (guoyongzhang) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go:231]: the next time maybe :-)

File: application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go (hunk 2 above)
Comment (guoyongzhang) [reply application-operator/controllers/wlsworkload/weblogicworkload_controller_test.go:214]: the next time maybe :-)

**Pull request merged**